src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -641,12 +657,7 @@ protected SymbolAndDiagnostics FindImplementationForInterfaceMemberWithDiagnosti
                         return SymbolAndDiagnostics.Empty;
                     }
 
-                    if (info.implementationForInterfaceMemberMap == null)
-                    {
-                        Interlocked.CompareExchange(ref info.implementationForInterfaceMemberMap, new ConcurrentDictionary<Symbol, SymbolAndDiagnostics>(), null);
-                    }
-
-                    return info.implementationForInterfaceMemberMap.GetOrAdd(interfaceMember, this.ComputeImplementationAndDiagnosticsForInterfaceMember);
+                    return info.ImplementationForInterfaceMemberMap.GetOrAdd(interfaceMember, this.ComputeImplementationAndDiagnosticsForInterfaceMember);
```

Comment:
> which is essentially what GetOrAdd does anyway

That's the same as long as there's no potential for a concurrent removal.  If there is, you'd need to do it in a loop.